src/sql/workbench/contrib/notebook/browser/htmlMarkdownConverter.ts

```diff
@@ -143,6 +143,36 @@ export class HTMLMarkdownConverter {
 				return `[${content}](${node.href})`;
 			}
 		});
+		// Only nested list case differs from original turndown rule
+		this.turndownService.addRule('list', {
+			filter: ['ul', 'ol'],
+			replacement: function (content, node) {
+				let parent = node.parentNode;
+				if ((parent.nodeName === 'LI' && parent.lastElementChild === node)) {
+					return '\n' + content;
+				} else if (parent.nodeName === 'UL' || parent.nodeName === 'OL') { // Nested list case
+					return '\n' + content + '\n';
+				} else {
+					return '\n\n' + content + '\n\n';
+				}
+			}
+		});
+		this.turndownService.addRule('lineBreak', {
+			filter: 'br',
+			replacement: function (content, node, options) {
+				if (node.parentElement?.nodeName !== 'LI') {
+					return options.br + '\n';
+				}
+				// One (and only one) line break is ignored when it's inside of a list item
```

Comment:
Maybe explain why - and in general more comments in the code would be useful since none of this stuff is very intuitive I think.

Reply:
In addition - when you say that only one line break is ignored do you mean that you skip the line break if there's only one? Or that it always skips the first one? Because it's doing the first right now but the comment seems to imply the second.